# Task-1 Making sense of the data

One of the reporters asks if you can find Mahmoud Khalil in the dataset. If you do, explain how you identified the correct record in the ICE data, by describing what logic you used, which fields you filtered on, how you handled any ambiguities and what steps you took to be confident it was the right match. 

## 0. Set up

In [1]:
from pathlib import Path
from datetime import datetime

import numpy as np
import pandas as pd

import process_data

In [2]:
arrests_filename = 'arrests-0923-0625.xlsx'
cwd = Path.cwd()
root = cwd.parent
data = root / "data"

In [3]:
arrests_df = process_data.read_arrests_data(data/arrests_filename)

In [5]:
arrests_df.head()

apprehension_date apprehension_state                   apprehension_aor  \
0 2024-08-07 09:43:00           VIRGINIA  WASHINGTON AREA OF RESPONSIBILITY   
1 2024-10-19 20:33:00              TEXAS     HOUSTON AREA OF RESPONSIBILITY   
2 2025-04-15 10:08:21         NEW JERSEY      NEWARK AREA OF RESPONSIBILITY   
3 2025-06-03 09:20:00          MINNESOTA    ST. PAUL AREA OF RESPONSIBILITY   
4 2025-01-21 17:41:00                NaN       MIAMI AREA OF RESPONSIBILITY   

                final_program        apprehension_method  \
0  ERO CRIMINAL ALIEN PROGRAM       NON-CUSTODIAL ARREST   
1  ERO CRIMINAL ALIEN PROGRAM    CAP LOCAL INCARCERATION   
2  ERO CRIMINAL ALIEN PROGRAM  CAP FEDERAL INCARCERATION   
3         FUGITIVE OPERATIONS       NON-CUSTODIAL ARREST   
4  ERO CRIMINAL ALIEN PROGRAM    CAP LOCAL INCARCERATION   

       apprehension_criminality                           case_status  \
0          1 CONVICTED CRIMINAL  8-EXCLUDED/REMOVED - INADMISSIBILITY   
1          1 CONVICTED CRIMINAL    6-DEPORTED/REMOVED - DEPORTABILITY   
2          1 CONVICTED CRIMINAL  8-EXCLUDED/REMOVED - INADMISSIBILITY   
3  3 OTHER IMMIGRATION VIOLATOR                                ACTIVE   
4    2 PENDING CRIMINAL CHARGES       3-VOLUNTARY DEPARTURE CONFIRMED   

                                       case_category departed_date  \
0                        [16] REINSTATED FINAL ORDER    2024-08-19   
1                        [16] REINSTATED FINAL ORDER    2024-10-22   
2                        [16] REINSTATED FINAL ORDER    2025-06-10   
3    [8G] EXPEDITED REMOVAL - CREDIBLE FEAR REFERRAL           NaT   
4  [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-02-01   

    departure_country final_order_yes_no  birth_year citizenship_country  \
0            HONDURAS                YES        1981            HONDURAS   
1              MEXICO                YES        1984              MEXICO   
2  DOMINICAN REPUBLIC                YES        1988  DOMINICAN REPUBLIC   
3                 NaN                YES        1985            COLOMBIA   
4              MEXICO                YES        1983              MEXICO   

   gender                         apprehension_site_landmark  \
0    MALE                     HBG GENERAL AREA, NON-SPECIFIC   
1    MALE                    HARRIS COUNTY JAIL, HOUSTON, TX   
2    MALE                          FORT DIX EAST, NEW JERSEY   
3  FEMALE                     SPM GENERAL AREA, NON-SPECIFIC   
4    MALE  MIAMI DADE COUNTY JAIL TURNER GUILFORD KNIGHT ...   

                          unique_identifier  
0  0000B34EDD657D516C02B13A7C352D62D0EFFCB6  
1  0000BA6E459998A6046D185D82CF4349DE1479D0  
2  0000C3D23FB0E444864559575900D410C4E8490F  
3  0000D3DBF8033B5F209F6547FFEE5B84FEB4F599  
4  000104D730BF021326C6DC0DEB3DD575304136B5

# Approach:

**Note:** for initial data exploration please see `./0-data_exploration.ipynb`

## 1. Choosing what to filter on:

**What information do we know about Mahmoud Khalil that we can use to find him in the dataset?**

Things that have been reported about Mahmoud Khalil:
* Location of arrest: New York (Columbia University Residence)
* Year of birth: 1995
* Citizenship: Algeria (born in Syria to Palestinian parents, and moved to Lebanon)
* He had a green card (ie not on a tempory visa or undocumented)
* Time and Date of Arrest: 2025-03-08, around 8.30pm 
* Where was he taken after his arrest? [his own understanding of this was shared in his letter](https://www.newsweek.com/mahmoud-khalil-columbia-hamas-gaza-israel-letter-2047002)
* * 26 Federal Plaza - on the 8th March (?)
  * Elizabeth Detention Centre, New Jersey - taken in the early hours of the 9th (NB when his wife tried to find him there on the 9th she was told he was not there, so there are some question marks over this)
  * ICE detention center in Louisiana 



### Finding the relevant variables:

In [36]:
birth_1995 = arrests_df['birth_year'] == 1995
citizenship_algeria = arrests_df['citizenship_country'] == 'ALGERIA' # might need to try out SYRIA and LEBANON here (although I don't think he has citizenship)

In [6]:
unique_vars_filter_str = lambda col, str_match: arrests_df[col][arrests_df[col].fillna('').str.contains(str_match)].unique()

#### Locations

##### 1. Arrest location = New York, initially taken to 26 Federal Plaza

In [7]:
np.sort(arrests_df['apprehension_state'].fillna('').unique())

array(['', 'ALABAMA', 'ALASKA', 'ARIZONA', 'ARKANSAS',
       'ARMED FORCES - EUROPE', 'ARMED FORCES - THE AMERICAS',
       'ARMED SERVICES - PACIFIC', 'CALIFORNIA', 'COLORADO',
       'CONNECTICUT', 'DELAWARE', 'DISTRICT OF COLUMBIA',
       'FEDERATED STATES OF MICRONESIA', 'FLORIDA', 'GEORGIA', 'GUAM',
       'HAWAII', 'IDAHO', 'ILLINOIS', 'INDIANA', 'IOWA', 'KANSAS',
       'KENTUCKY', 'LOUISIANA', 'MAINE', 'MARYLAND', 'MASSACHUSETTS',
       'MEXICO', 'MICHIGAN', 'MINNESOTA', 'MISSISSIPPI', 'MISSOURI',
       'MONTANA', 'NEBRASKA', 'NEVADA', 'NEW HAMPSHIRE', 'NEW JERSEY',
       'NEW MEXICO', 'NEW YORK', 'NORTH CAROLINA', 'NORTH DAKOTA',
       'NORTHERN MARIANA ISLANDS', 'OHIO', 'OKLAHOMA', 'OREGON',
       'PENNSYLVANIA', 'PUERTO RICO', 'RHODE ISLAND', 'SOUTH CAROLINA',
       'SOUTH DAKOTA', 'TAMAULIPAS', 'TENNESSEE', 'TEXAS', 'UTAH',
       'VERMONT', 'VIRGIN ISLANDS', 'VIRGINIA', 'WASHINGTON',
       'WEST VIRGINIA', 'WISCONSIN', 'WYOMING'], dtype=object)

In [8]:
unique_vars_filter_str('apprehension_site_landmark','26 FEDERAL')

array(['NDD - 26 FEDERAL PLAZA NY, NY'], dtype=object)

In [9]:
arrests_df[arrests_df['apprehension_site_landmark']=='NDD - 26 FEDERAL PLAZA NY, NY']['apprehension_state'].value_counts(dropna=False)

apprehension_state
NEW YORK          5648
NaN                151
NORTH CAROLINA       4
NEW JERSEY           1
FLORIDA              1
Name: count, dtype: int64



**Observations:**
* Looks like that is the correct site for 26 Federal Plaza, NY
* But note - not all who have `'apprehension_site_landmark' == 'NDD - 26 FEDERAL PLAZA NY, NY'` have `'apprehension_state' == 'NEW YORK'`
   * Maybe taken to be processed at 26 Federal Plaza after being initially arrested in the location? But Florida and North Carolina are quite a long way
   * 151 with missing `apprehension_state`

In [37]:
state_ny = arrests_df['apprehension_state'] == 'NEW YORK' # NB - from initial data exploration we know this has quite a lot of missing, so don't rely on it being added
loc_federal_plaza = arrests_df['apprehension_site_landmark'] == 'NDD - 26 FEDERAL PLAZA NY, NY'

##### 2. Taken to: Elizabeth Detention Center in New Jersey and ICE Detention Center in Louisiana

Doesn't look like these are in this dataset, which makes sense because this is arrests, not detentions:

In [10]:
unique_vars_filter_str('apprehension_site_landmark','ELIZABETH')

array(['ELIZABETH CITY', 'ELIZABETH PD'], dtype=object)

In [11]:
unique_vars_filter_str('apprehension_site_landmark','LOUISIANA')

array(['LOUISIANA STATE PENITENTIARY - ANGOLA'], dtype=object)

#### 2. Apprehension date

In [28]:
get_date = lambda date_str: datetime.strptime(date_str, '%Y-%m-%d').date()

In [34]:
def filter_dates(df, start_date, eq=True, end_date=None, date_col='apprehension_date'):
    if end_date:
        return (df[date_col].dt.date >= get_date(start_date)) & (df[date_col].dt.date <= get_date(end_date))
    elif eq:
        return df[date_col].dt.date == get_date(start_date)
    else:
        return df[date_col].dt.date >= get_date(start_date)

In [54]:
arrests_df[filter_dates(arrests_df, '2025-03-08')]

apprehension_date apprehension_state  \
429    2025-03-08 22:05:00              TEXAS   
1801   2025-03-08 08:53:00            FLORIDA   
4209   2025-03-08 23:36:00            GEORGIA   
6619   2025-03-08 09:47:00         WASHINGTON   
6848   2025-03-08 07:13:00              TEXAS   
...                    ...                ...   
264332 2025-03-08 04:39:00              TEXAS   
264333 2025-03-08 09:50:11              TEXAS   
264334 2025-03-08 07:54:37              TEXAS   
264335 2025-03-08 09:47:51              TEXAS   
264336 2025-03-08 09:51:28              TEXAS   

                        apprehension_aor                final_program  \
429       HOUSTON AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
1801        MIAMI AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
4209      ATLANTA AREA OF RESPONSIBILITY                 287G PROGRAM   
6619      SEATTLE AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
6848    HARLINGEN AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
...                                  ...                          ...   
264332     DALLAS AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
264333  HARLINGEN AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
264334  HARLINGEN AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
264335  HARLINGEN AREA OF RESPONSIBILITY      DETAINED DOCKET CONTROL   
264336  HARLINGEN AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   

              apprehension_method      apprehension_criminality  \
429       CAP LOCAL INCARCERATION    2 PENDING CRIMINAL CHARGES   
1801                      LOCATED  3 OTHER IMMIGRATION VIOLATOR   
4209               287(G) PROGRAM    2 PENDING CRIMINAL CHARGES   
6619                      LOCATED          1 CONVICTED CRIMINAL   
6848    CAP FEDERAL INCARCERATION  3 OTHER IMMIGRATION VIOLATOR   
...                           ...                           ...   
264332       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
264333    CAP LOCAL INCARCERATION  3 OTHER IMMIGRATION VIOLATOR   
264334    CAP LOCAL INCARCERATION    2 PENDING CRIMINAL CHARGES   
264335                    LOCATED    2 PENDING CRIMINAL CHARGES   
264336    CAP LOCAL INCARCERATION    2 PENDING CRIMINAL CHARGES   

                                 case_status  \
429     8-EXCLUDED/REMOVED - INADMISSIBILITY   
1801    8-EXCLUDED/REMOVED - INADMISSIBILITY   
4209    8-EXCLUDED/REMOVED - INADMISSIBILITY   
6619      6-DEPORTED/REMOVED - DEPORTABILITY   
6848      6-DEPORTED/REMOVED - DEPORTABILITY   
...                                      ...   
264332                                   NaN   
264333                        9-VR WITNESSED   
264334                        9-VR WITNESSED   
264335                        9-VR WITNESSED   
264336                        9-VR WITNESSED   

                                            case_category departed_date  \
429     [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-05-03   
1801                          [16] REINSTATED FINAL ORDER    2025-03-18   
4209    [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-05-12   
6619                          [16] REINSTATED FINAL ORDER    2025-06-03   
6848                          [16] REINSTATED FINAL ORDER    2025-03-12   
...                                                   ...           ...   
264332                                                NaN           NaT   
264333                            [9] VR UNDER SAFEGUARDS    2025-03-09   
264334                            [9] VR UNDER SAFEGUARDS    2025-03-08   
264335                            [9] VR UNDER SAFEGUARDS    2025-03-08   
264336                            [9] VR UNDER SAFEGUARDS    2025-03-08   

       departure_country final_order_yes_no  birth_year citizenship_country  \
429            GUATEMALA                YES        2002           GUATEMALA   
1801           GUATEMALA                YES        1988           GUATEMALA   
4209           NICARAGU

#### 3. Immigration status

According to the documentation, Unique Identifier is based on Alien Registration Number (A-number). As Mahmoud Khalil had a green card, we might assume he would not have an A-number, and therefore wouldn't have a Unique Identifier. This is a big assumption though, and the ICE enforcement agents did think he had a student visa, so he might have an A-number from before he had a green card?

In [64]:
no_ui = arrests_df['unique_identifier'].isna()

Are there any other clues in other variables that indicate immigration status?

In [65]:
arrests_df['final_program'].value_counts()

final_program
ERO CRIMINAL ALIEN PROGRAM             162433
FUGITIVE OPERATIONS                     39776
NON-DETAINED DOCKET CONTROL             33165
ALTERNATIVES TO DETENTION               12761
DETAINED DOCKET CONTROL                  8972
287G PROGRAM                             4934
MOBILE CRIMINAL ALIEN TEAM               2214
JUVENILE                                  444
ERO CRIMINAL PROSECUTIONS                 304
DETENTION AND DEPORTATION                 159
LAW ENFORCEMENT AREA RESPONSE UNITS        45
287G TASK FORCE                            10
VIOLENT CRIMINAL ALIEN SECTION              5
JOINT CRIMINAL ALIEN RESPONSE TEAM          4
Name: count, dtype: int64

In [66]:
arrests_df['apprehension_method'].value_counts()

apprehension_method
CAP LOCAL INCARCERATION                        112103
NON-CUSTODIAL ARREST                            56910
LOCATED                                         31405
CAP FEDERAL INCARCERATION                       23408
CAP STATE INCARCERATION                         10419
OTHER EFFORTS                                    9018
ERO REPROCESSED ARREST                           8875
287(G) PROGRAM                                   6313
PROBATION AND PAROLE                             3833
LAW ENFORCEMENT AGENCY RESPONSE UNIT              845
OTHER TASK FORCE                                  505
PATROL BORDER                                     477
OTHER AGENCY (TURNED OVER TO INS)                 444
WORKSITE ENFORCEMENT                              252
INSPECTIONS                                       127
ANTI-SMUGGLING                                     84
TRAFFIC CHECK                                      62
ORGANIZED CRIME DRUG ENFORCEMENT TASK FORCE        52
PATROL I

Potentially useful info in Final Program, but would have to do some more research to clarify meaning of some of the categories

## 2. Filtering - trying to find Mahmoud Khalil

### 1. Location and date:

#### In 26 Federal Plaza around 8th March 2025:

In [68]:
arrests_df[filter_dates(arrests_df,'2025-03-07', end_date='2025-03-10') & loc_federal_plaza]

The history saving thread hit an unexpected error (OperationalError('attempt to write a readonly database')).History will not be written to the database.


apprehension_date apprehension_state  \
66791  2025-03-09 11:15:00           NEW YORK   
90940  2025-03-10 10:27:00           NEW YORK   
136002 2025-03-10 08:58:00           NEW YORK   
181085 2025-03-10 08:49:00           NEW YORK   

                            apprehension_aor                final_program  \
66791   NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
90940   NEW YORK CITY AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
136002  NEW YORK CITY AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
181085  NEW YORK CITY AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   

         apprehension_method      apprehension_criminality  \
66791   NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
90940   NON-CUSTODIAL ARREST    2 PENDING CRIMINAL CHARGES   
136002  NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
181085  NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   

                                 case_status  \
66791                                 ACTIVE   
90940   8-EXCLUDED/REMOVED - INADMISSIBILITY   
136002    6-DEPORTED/REMOVED - DEPORTABILITY   
181085                                ACTIVE   

                                            case_category departed_date  \
66791          [2A] DEPORTABLE - UNDER ADJUDICATION BY IJ           NaT   
90940   [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-04-22   
136002                        [16] REINSTATED FINAL ORDER    2025-03-18   
181085  [8K] EXPEDITED REMOVAL TERMINATED DUE TO CREDI...           NaT   

       departure_country final_order_yes_no  birth_year citizenship_country  \
66791                NaN                 NO        1973            HONDURAS   
90940            ECUADOR                YES        1988             ECUADOR   
136002           ECUADOR                YES        1987             ECUADOR   
181085               NaN                 NO        1985          BANGLADESH   

        gender     apprehension_site_landmark  \
66791   FEMALE  NDD - 26 FEDERAL PLAZA NY, NY   
90940   FEMALE  NDD - 26 FEDERAL PLAZA NY, NY   
136002    MALE  NDD - 26 FEDERAL PLAZA NY, NY   
181085    MALE  NDD - 26 FEDERAL PLAZA NY, NY   

                               unique_identifier  
66791   40F5826248C65DF902098458BC301DCCA7A30C14  
90940   589109DAF90CD9AF35E0A5F2EEE32E2ABCF9EB2F  
136002  84443C7BC8DC8C9CD34EB762CD8F5A8E2AA1106F  
181085  B0123DFB853895BB64FEABD32E1B7781F5330E40

#### In New York around 8th March 2025:

In [71]:
arrests_df[filter_dates(arrests_df,'2025-03-07', end_date='2025-03-10') & state_ny].sort_values(by='apprehension_date')

apprehension_date apprehension_state  \
142348 2025-03-07 10:07:00           NEW YORK   
17660  2025-03-07 10:12:28           NEW YORK   
172444 2025-03-07 10:54:00           NEW YORK   
116429 2025-03-07 10:59:00           NEW YORK   
178822 2025-03-08 00:23:00           NEW YORK   
162513 2025-03-08 09:18:00           NEW YORK   
73250  2025-03-09 10:52:00           NEW YORK   
66791  2025-03-09 11:15:00           NEW YORK   
169123 2025-03-09 11:15:00           NEW YORK   
60359  2025-03-09 15:13:00           NEW YORK   
227318 2025-03-09 18:21:00           NEW YORK   
158918 2025-03-10 08:40:00           NEW YORK   
181085 2025-03-10 08:49:00           NEW YORK   
136002 2025-03-10 08:58:00           NEW YORK   
161582 2025-03-10 09:23:00           NEW YORK   
171212 2025-03-10 09:46:00           NEW YORK   
90940  2025-03-10 10:27:00           NEW YORK   
35693  2025-03-10 10:46:00           NEW YORK   

                            apprehension_aor                final_program  \
142348        BUFFALO AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
17660         BUFFALO AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
172444  NEW YORK CITY AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
116429  NEW YORK CITY AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
178822  NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
162513        BUFFALO AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
73250         BUFFALO AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
66791   NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
169123  NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
60359   NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
227318  NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
158918  NEW YORK CITY AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
181085  NEW YORK CITY AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
136002  NEW YORK CITY AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
161582         NEWARK AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
171212        BUFFALO AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
90940   NEW YORK CITY AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
35693         BUFFALO AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   

              apprehension_method      apprehension_criminality  \
142348       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
17660     CAP STATE INCARCERATION          1 CONVICTED CRIMINAL   
172444  CAP FEDERAL INCARCERATION          1 CONVICTED CRIMINAL   
116429       NON-CUSTODIAL ARREST          1 CONVICTED CRIMINAL   
178822                    LOCATED    2 PENDING CRIMINAL CHARGES   
162513       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
73250        NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
66791        NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
169123       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
60359                     LOCATED  3 OTHER IMMIGRATION VIOLATOR   
227318       NON-CUSTODIAL ARREST    2 PENDING CRIMINAL CHARGES   
158918       PROBATION AND PAROLE          1 CONVICTED CRIMINAL   
181085       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
136002       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
161582       NON-CUSTODIAL ARREST    2 PENDING CRIMINAL CHARGES   
171212       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
90940        NON-CUSTODIAL ARREST    2 PENDING CRIMINAL CHARGES   
35693        NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   

                                 case_status  \
142348  8-EXCLUDED/REMOVED - INADMISSIBILITY   
17660                                 ACTIVE   
172444                                ACTIVE   
116429  8-EXCLUDED/REMOVED - INADMISSIBILITY   
178822                                ACTIVE   
162513  8-EXCLUDED/REMOVED - INADMISSIBILITY   
732

There do not seem to be a clear match for either location field around the timeframe. 

### 2. Demographics

#### Citizenship == Algeria, and Birth Year is 1995:

In [72]:
arrests_df[citizenship_algeria & birth_1995]

apprehension_date apprehension_state                apprehension_aor  \
107560 2025-03-11 12:56:29           NEW YORK  BUFFALO AREA OF RESPONSIBILITY   

                     final_program      apprehension_method  \
107560  ERO CRIMINAL ALIEN PROGRAM  CAP STATE INCARCERATION   

       apprehension_criminality case_status  \
107560     1 CONVICTED CRIMINAL      ACTIVE   

                                     case_category departed_date  \
107560  [2A] DEPORTABLE - UNDER ADJUDICATION BY IJ           NaT   

       departure_country final_order_yes_no  birth_year citizenship_country  \
107560               NaN                 NO        1995             ALGERIA   

       gender       apprehension_site_landmark  \
107560   MALE  SING SING CORRECTIONAL FACILITY   

                               unique_identifier  
107560  68BC0E521695985B7AFB1AE3D8A2F53A2E75F572

#### Could this be Mahmoud Khalil?

**Headline:** I think it is unlikely. While there are some clear similarities, there are important inconsistancies that don't fit what is generally accepted about the case.

Evidence in support:
* Apprehension in New York close to the time Mahmoud Khalil was arrested
* Demographic match - Citizenship is Algerian, and Birth Year is 1995

Evidence against:
* Apprehension site landmark is Sing Sing Correctional Facility, which is a prison in upstate New York. There is no reporting or other evidence to suggest Mahmoud Khalil was taken to Sing Sing, and the timeframe does not really allow for it (he was known to be in Louisiana around the same time)
* Apprehension method is "CAP STATE INCARCERATION" - CAP is Criminal Alien Program ([source: ICE.gov](https://www.ice.gov/identify-and-arrest/criminal-alien-program)), and "CAP STATE INCARCERATION" suggests this is an example of someone already incarcerated being arrested by ICE 
